frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerLifeCycle.java

```diff
@@ -335,10 +339,11 @@ public void run() {
                         continue;
                     }
 
-                    if ("Torch worker started.".equals(result)) {
+                    Matcher matcher = PID_LOG_PATTERN.matcher(result);
+                    if (result.endsWith("Torch worker started.")) {
                         lifeCycle.setSuccess(true);
-                    } else if (result.startsWith("[PID]")) {
-                        lifeCycle.setPid(Integer.parseInt(result.substring("[PID]".length())));
+                    } else if (matcher.matches()) {
+                        lifeCycle.setPid(Integer.parseInt(matcher.group(1)));
```

Comment:
Why log record ending with "Torch worker started." not work in cpp?

Reply:
Since we are using folly for backend logging the only out of the box formatter is glog ([doc](https://github.com/facebook/folly/blob/main/folly/logging/docs/LogHandlers.md#formatter)). That adds some additional info to the beginning of the log line which has to be accounted for